screenpipe-server/src/server.rs

```diff
@@ -1571,6 +1571,59 @@ async fn get_similar_speakers_handler(
     Ok(JsonResponse(similar_speakers))
 }
 
+// Add these new structs after the other request/response structs
+#[derive(Deserialize)]
+pub struct AudioDeviceControlRequest {
+    device_name: String,
+    #[serde(default)]
+    device_type: Option<DeviceType>,
+}
+
+#[derive(Serialize)]
+pub struct AudioDeviceControlResponse {
+    success: bool,
+    message: String,
+}
+
+// Add these new handler functions before create_router()
+async fn start_audio_device(
+    State(state): State<Arc<AppState>>,
+    Json(payload): Json<AudioDeviceControlRequest>,
+) -> Result<JsonResponse<AudioDeviceControlResponse>, (StatusCode, JsonResponse<Value>)> {
+    let device = AudioDevice {
+        name: payload.device_name.clone(),
+        device_type: payload.device_type.unwrap_or(DeviceType::Input),
+    };
+
+    state.audio_devices_control.push((device.clone(), DeviceControl { is_running: true, is_paused: false }));
```

Comment:
does this work? i implemented audio_devices_control back then but not sure it's even working anymore